assets/script/script.js

```diff
@@ -5,39 +5,60 @@ let weatherWord;
+  //iterate through local storage history
+  for (let i = 0; i < history.length; i++) {
+    //passing the history data of local storage to a function named display History
+    displayHist(history[i]);
+  }
+
+  $("#historyList").on("click", "li", function () {
+    city = $(this).text();
+    apiCalls(city);
+  });
   // listen for city input
   $("#searchBtn").on("click", function () {
-    $("#weather-music-list").empty();
+    //$("#weather-music-list").empty();
     city = $("#citySearch").val();
-    // get weather word for current weather in that city
-    let weatherApiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=imperial&appid=${OPENWEATHER_API_KEY}`;
+    if (city) {
+      apiCalls(city);
+      if (history.indexOf(city) === -1) {
+        history.push(city);
+        window.localStorage.setItem("history", JSON.stringify(history));
+        displayHist(city);
+      }
+    }
+  });
+});
+
+function apiCalls(city) {
+  let weatherApiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=imperial&appid=${OPENWEATHER_API_KEY}`;
+  $.ajax({
+    url: weatherApiUrl,
+    method: "GET",
+  }).then(function (response) {
+    weatherWord = response.weather[0].main;
+    displayWeather(response);
+    console.log(`weatherWord = ${weatherWord}, city = ${city}`);
+    let queryString = `?part=snippet&maxResults=25&q=${weatherWord}%20Music&type=playlist&key=${YOUTUBE_API_KEY}`;
+    let youtubeAPIURL = YOUTUBE_SEARCH_ENDPOINT + queryString;
     $.ajax({
-      url: weatherApiUrl,
+      url: youtubeAPIURL,
       method: "GET",
     }).then(function (response) {
-      weatherWord = response.weather[0].main;
-      displayWeather(response);
-      console.log(`weatherWord = ${weatherWord}, city = ${city}`);
-      let queryString = `?part=snippet&maxResults=25&q=${weatherWord}%20Music&type=playlist&key=${YOUTUBE_API_KEY}`;
-      let youtubeAPIURL = YOUTUBE_SEARCH_ENDPOINT + queryString;
-      $.ajax({
-        url: youtubeAPIURL,
-        method: "GET",
-      }).then(function (response) {
-        console.log(response);
-        // get a random playlist in the response
-        let randomInt = Math.floor(Math.random() * response.items.length);
-        let randomPlaylistId = response.items[randomInt].id.playlistId;
-        playlistId = randomPlaylistId;
-        createIframe(playlistId);
-      });
+      console.log(response);
+      // get a random playlist in the response
+      let randomInt = Math.floor(Math.random() * response.items.length);
+      let randomPlaylistId = response.items[randomInt].id.playlistId;
+      playlistId = randomPlaylistId;
+      createIframe(playlistId);
     });
   });
-});
+}
 
 function displayWeather(response) {
-  let cityName = $(".nameTemp").html(response.name + " " + response.main.temp + " °F");
-  let weatherDay = $(".weatherDay").html(`It's a ${weatherWord} day`);
+  $(".nameTemp").html(response.name + " " + response.main.temp + " °F");
```

Comment:
I think the `.html` here can be changed to `.text(response.name...`.

Reply:
It's working, so we can leave it for now, but I think `.html` is typically used for constructing some html code within the selected element.